sdks/python/apache_beam/runners/worker/bundle_processor.py

```diff
@@ -313,6 +313,14 @@ def metrics(self):
             self._fix_output_tags(transform_id, op.progress_metrics())
             for transform_id, op in self.ops.items()})
 
+  def monitoring_infos(self):
```

Comment:
Curious about reviewer thoughts on how to test this best. Would it make sense to test BundleProcessor on its own? It might be a bit complex to set this up, not sure if this is the kind of testing approach that we want for beam.

Currently I have tested this indirectly via fn_api_runner_test (this is how the existing user metrics were being tested, the 'system' element count and processing time metrics were not tested at all :( )

Reply:
First approach would be preferred, if this will be a large burden it is fine to keep the existing setup.

Reply:
Since I was not able to properly instantiate these classes I have tested the monitoring_infos by providing an accessor to the MetricResults and updated the fn_api_runner_test accordingly. As discussed in this thread:

https://lists.apache.org/thread.html/6ad0d5446ffc08850ef684e3fc4e0c2d39ef58e9bad2b8419db02b41@%3Cdev.beam.apache.org%3E

Reply:
Testing the bundle processor requires providing a properly formed execution tree. The easiest way by far to construct such a tree is by using the user-facing API (which, for the moment, is also significantly more stable). This is also in line with the mantra of testing user-facing behavior rather than implementation details. 

Of course sometimes one needs more focused implementation detail tests.